src/ssh_agent.rs

```diff
@@ -233,15 +234,15 @@ impl<
             }
         };
 
-        match key.private_key {
+        match key.private_key() {
             Some(private_key) => {
                 writebuf.push(msg::SIGN_RESPONSE);
-                let signer: &dyn SigningKey = &private_key;
-                let sig = signer.try_sign(data).or(Err(SSHAgentError::AgentFailure));
+                let sig = private_key
+                    .try_sign(data)
+                    .or(Err(SSHAgentError::AgentFailure));
                 let sig = match sig {
                     Ok(sig) => sig,
-                    Err(err) => {
-                        println!("Error signing: {:?}", err);
+                    Err(_) => {
                         writebuf.push(msg::FAILURE);
                         return Ok((agent, false));
                     }
```

Comment:
Maybe clippy will complain, but we might be able to simplify this with a let-else? Maybe not. Ex:

```rust
let Ok(sig) = sig else {
  // err stuff here
}

// sig should be unwrapped here
```

Reply:
Applied!